src/python/pants/init/engine_initializer.py

```diff
@@ -371,6 +374,20 @@ def union_membership_singleton() -> UnionMembership:
     def build_root_singleton() -> BuildRoot:
       return BuildRoot.instance
 
+    @rule
+    def single_build_file_address(specs: Specs) -> BuildFileAddress:
+      build_file_addresses = yield Get(BuildFileAddresses, Specs, specs)
+      if len(build_file_addresses.dependencies) != 1:
```

Comment:
This conditional makes the error message incorrect if the spec doesn't match anything.

Reply:
Is there a way to trigger this? If a spec is a valid build file that has no runnable targets, it throws a separate ResolveError,and if the spec isn't a build file at all it throws a different ResolveError.

Reply:
```$ mkdir blah
$ touch blah/BUILD
$ ./pants list blah/:
WARNING: No targets were matched in goal `list`.
$ echo $?
0```